### PR TITLE
Update azure-identity docs

### PR DESCRIPTION
**Pull request by chlowell**
* adds ref doc links to the readme and revises its content
* adds package-level docstrings for the benefit of docs.microsoft.com
* adds a note on every `get_token` to discourage customers from thinking they must use it
* updates history

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/identity/azure-identity/README.md [hunk 1]
```diff
@@ -26,99 +24,102 @@ pip install azure-identity
...
+>|`AZURE_TENANT_ID`|id of the application's Azure Active Directory tenant
+>|`AZURE_CLIENT_SECRET`|one of the application's client secrets
 
 #### Service principal with certificate
 >|variable name|value
 >|-|-
->|`AZURE_CLIENT_ID`|service principal's app id
->|`AZURE_TENANT_ID`|id of the principal's Azure Active Directory tenant
->|`AZURE_CLIENT_CERTIFICATE_PATH`|path to a PEM-encoded certificate file including private key (without password)
+>|`AZURE_CLIENT_ID`|id of an Azure Active Directory application
```
Comment (schaabs) [inline sdk/identity/azure-identity/README.md]: So there's nuance here, and I'm not sure how we should highlight it here. This technically should be service principal id. In most cases the application id and the SP's id should be the same. But if the app is a multi-tenant app they wouldn't be.

Comment (chlowell) [reply sdk/identity/azure-identity/README.md]: I struggle with what to call this. People with working knowledge of AAD are fine with "service principal", which I agree is more correct. However, "application" is used all over the place to refer to the same thing. The portal's and CLI's usage of it led me to make this change, hoping "application" can split the difference, helping new users find the right value without confusing the experienced.

Comment (schaabs) [reply sdk/identity/azure-identity/README.md]: Your probably right that here in the high level view we should avoid exposing this nuance as multi-tenant apps are not the common case (and not the case we have in the creation commands above). So I like you're change here, but at some point we probably need deeper level docs that goes into some of the nuance.

File: sdk/identity/azure-identity/README.md [hunk 2]
```diff
@@ -26,99 +24,102 @@ pip install azure-identity
...
 
-To authenticate as a service principal, provide configuration in environment
-variables as described in the next section.
+To authenticate as a service principal, provide configuration in
+[environment variables](#environment-variables) as described in the next section.
 
-Authenticating as a managed identity requires no configuration, but does
-require platform support. See the
+Authenticating as a managed identity requires no configuration but is only
+possible on a supported platform. See Azure Active Directory's
```
[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/identity/azure-identity/README.md]: Thanks, that's a good point, let's try ~on a supported platform~ "in a supported hosting environment".

**Pull request merged**